src/lxc/utils.c

```diff
@@ -910,9 +910,24 @@ char *get_template_path(const char *t)
 	int ret, len;
 	char *tpath;
 
-	if (t[0] == '/' && access(t, X_OK) == 0) {
-		tpath = strdup(t);
-		return tpath;
+	if (t[0] == '/') {
+        	if (access(t, X_OK) == 0) {
+			tpath = strdup(t);
+			return tpath;
+		} else {
+			if (errno == EACCES) {
+				SYSERROR("cannot execute template %s", t);
+				return NULL;
+			}
+
+			if (errno == ENOENT) {
+				SYSERROR("template %s does not exist", t);
+				return NULL;
+			}
+
+			SYSERROR("Bad template pathname: %s", t);
+			return NULL;
+		}
 	}
```

Comment:
Why doesn't a single `SYSERROR` suffice? The errno string will be printed and hence does give you the root cause. So I'd say just do:
```
SYSERROR("Failed to execute template");
```